File: archive/feature/library/src/main/java/com/jian/nemo/feature/library/presentation/category/CategoryClassificationScreen.kt
package com.jian.nemo.feature.library.presentation.category

import androidx.compose.animation.core.Animatable
import androidx.compose.animation.core.tween
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.rounded.ArrowBack
import androidx.compose.material.icons.automirrored.rounded.ArrowForward
import androidx.compose.material.icons.automirrored.rounded.DirectionsRun
import androidx.compose.material.icons.automirrored.rounded.MenuBook
import androidx.compose.material.icons.automirrored.rounded.Sort
import androidx.compose.material.icons.rounded.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.jian.nemo.core.designsystem.theme.NemoCategoryColors
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

/**
 * 词性分类主题颜色
 */
data class CategoryThemeColor(
    val containerColor: Color,
    val contentColor: Color
)

/**
 * 根据当前主题获取分类颜色
 */
@Composable
private fun getCategoryTheme(
    lightContainer: Color,
    lightContent: Color,
    darkContainer: Color,
    darkContent: Color
): CategoryThemeColor {
    // 保持原来的颜色逻辑用于Icon
    val isDark = MaterialTheme.colorScheme.surface.luminance < 0.5
    return CategoryThemeColor(
        containerColor = if (isDark) darkContainer else lightContainer,
        contentColor = if (isDark) darkContent else lightContent
    )
}

/**
 * 专项训练 - 分类选择界面
 *
 * 风格统一：
 * - Background: Solid (MaterialTheme.colorScheme.background)
 * - Card: Premium Card (White/SurfaceContainer + Shadow + 26dp Rounded)
 * - Icon: Squircle (Rounded 14dp)
 * - Typography: ExtraBold Titles
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun CategoryClassificationScreen(
    source: String = "practice",
    onNavigateBack: () -> Unit = {},
    onCategorySelected: (String, String) -> Unit = { _, _ -> }
) {
    val backgroundColor = MaterialTheme.colorScheme.background

    // 根据来源显示不同的标题
    val title = when (source) {
        "practice" -> "专项训练"
        "vocabulary" -> "专项词汇"
        else -> "专项分类"
    }

    Scaffold(
        topBar = {
            com.jian.nemo.core.ui.component.common.CommonHeader(
                title = title,
                onBack = onNavigateBack,
                backgroundColor = backgroundColor // Header 背景与页面一致
            )
        },
        containerColor = backgroundColor
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .verticalScroll(rememberScrollState())
                .padding(horizontal = 20.dp, vertical = 24.dp)
        ) {
            // 1. 基础词性类
            SectionHeader(title = "基础词性类", delay = 0)
            Spacer(modifier = Modifier.height(16.dp))
            CategoryGrid(
                items = listOf(
                    CategoryItem(
                        id = "noun",
                        title = "名词类",
                        subtitle = "名词、代词等",
                        icon = Icons.AutoMirrored.Rounded.MenuBook,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardNounBgLight, NemoCategoryColors.CardNounTextLight,
                            NemoCategoryColors.CardNounBgDark, NemoCategoryColors.CardNounTextDark
                        ) }
                    ),
                    CategoryItem(
                        id = "adj",
                        title = "形容词类",
                        subtitle = "い形、な形形容词",
                        icon = Icons.Rounded.Stars,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardAdjIBgLight, NemoCategoryColors.CardAdjITextLight,
                            NemoCategoryColors.CardAdjIBgDark, NemoCategoryColors.CardAdjITextDark
                        ) }
                    ),
                    CategoryItem(
                        id = "verb",
                        title = "动词类",
                        subtitle = "自动/他动/自他動词",
                        icon = Icons.AutoMirrored.Rounded.DirectionsRun,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardVerbBgLight, NemoCategoryColors.CardVerbTextLight,
                            NemoCategoryColors.CardVerbBgDark, NemoCategoryColors.CardVerbTextDark
                        ) }
                    ),
                    CategoryItem(
                        id = "adv",
                        title = "副词",
                        subtitle = "修饰用言词汇",
                        icon = Icons.AutoMirrored.Rounded.Sort,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardAdvBgLight, NemoCategoryColors.CardAdvTextLight,
                            NemoCategoryColors.CardAdvBgDark, NemoCategoryColors.CardAdvTextDark
                        ) }
                    )
                ),
                onItemClick = { category ->
                    onCategorySelected(category.id, category.title)
                }
            )

            Spacer(modifier = Modifier.height(32.dp))

            // 2. 构词·句法功能类
            SectionHeader(title = "構词·句法功能类", delay = 100)
            Spacer(modifier = Modifier.height(16.dp))
            CategoryGrid(
                items = listOf(
                    CategoryItem(
                        id = "rentai",
                        title = "连体词",
                        subtitle = "直接修饰体言",
                        icon = Icons.Rounded.Link,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardRentaiBgLight, NemoCategoryColors.CardRentaiTextLight,
                            NemoCategoryColors.CardRentaiBgDark, NemoCategoryColors.CardRentaiTextDark
                        ) }
                    ),
                    CategoryItem(
                        id = "conj",
                        title = "接続词",
                        subtitle = "连接句子成分",
                        icon = Icons.Rounded.LinearScale,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardConjBgLight, NemoCategoryColors.CardConjTextLight,
                            NemoCategoryColors.CardConjBgDark, NemoCategoryColors.CardConjTextDark
                        ) }
                    ),
                    CategoryItem(
                        id = "exclam",
                        title = "感叹词",
                        subtitle = "表达情感语气",
                        icon = Icons.Rounded.Campaign,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardIdiomBgLight, NemoCategoryColors.CardIdiomTextLight,
                            NemoCategoryColors.CardIdiomBgDark, NemoCategoryColors.CardIdiomTextDark
                        ) }
                    ),
                    CategoryItem(
                        id = "particle",
                        title = "助词",
                        subtitle = "语法功能标记",
                        icon = Icons.Rounded.Attribution,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardFixBgLight, NemoCategoryColors.CardFixTextLight,
                            NemoCategoryColors.CardFixBgDark, NemoCategoryColors.CardFixTextDark
                        ) }
                    )
                ),
                onItemClick = { category ->
                    onCategorySelected(category.id, category.title)
                }
            )

            Spacer(modifier = Modifier.height(32.dp))

            // 3. 构词·表达用法类
            SectionHeader(title = "構词·表达用法类", delay = 200)
            Spacer(modifier = Modifier.height(16.dp))
            CategoryGrid(
                items = listOf(
                    CategoryItem(
                        id = "prefix",
                        title = "接头词",
                        subtitle = "词语前置构成",
                        icon = Icons.AutoMirrored.Rounded.ArrowForward,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardKataBgLight, NemoCategoryColors.CardKataTextLight,
                            NemoCategoryColors.CardKataBgDark, NemoCategoryColors.CardKataTextDark
                        ) }
                    ),
                    CategoryItem(
                        id = "suffix",
                        title = "接尾词",
                        subtitle = "词语后置构成",
                        icon = Icons.AutoMirrored.Rounded.ArrowBack,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardSoundBgLight, NemoCategoryColors.CardSoundTextLight,
                            NemoCategoryColors.CardSoundBgDark, NemoCategoryColors.CardSoundTextDark
                        ) }
                    ),
                    CategoryItem(
                        id = "expression",
                        title = "表达·固定句型",
                        subtitle = "习惯表达方式",
                        icon = Icons.Rounded.FormatQuote,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardKeigoBgLight, NemoCategoryColors.CardKeigoTextLight,
                            NemoCategoryColors.CardKeigoBgDark, NemoCategoryColors.CardKeigoTextDark
                        ) }
                    ),
                    CategoryItem(
                        id = "kata",
                        title = "外来语",
                        subtitle = "片假名借词体系",
                        icon = Icons.Rounded.Language,
                        colors = { getCategoryTheme(
                            NemoCategoryColors.CardAdjNaBgLight, NemoCategoryColors.CardAdjNaTextLight,
                            NemoCategoryColors.CardAdjNaBgDark, NemoCategoryColors.CardAdjNaTextDark
                        ) }
                    )
                ),
                onItemClick = { category ->
                    onCategorySelected(category.id, category.title)
                }
            )

            Spacer(modifier = Modifier.height(32.dp))
        }
    }
}

@Composable
private fun SectionHeader(title: String, delay: Int = 0) {
    val alpha = remember { Animatable(0f) }

    LaunchedEffect(Unit) {
        delay(delay.toLong())
        launch { alpha.animateTo(1f, tween(500)) }
    }

    // Style matches LearningCalendarSectionTitle
    Text(
        text = title,
        style = MaterialTheme.typography.titleMedium.copy(
            fontWeight = FontWeight.ExtraBold,
            letterSpacing = 0.5.sp
        ),
        color = MaterialTheme.colorScheme.onSurfaceVariant,
        modifier = Modifier
            .fillMaxWidth()
            .alpha(alpha.value)
            .padding(horizontal = 4.dp, vertical = 4.dp)
    )
}

@Composable
private fun CategoryGrid(
    items: List<CategoryItem>,
    onItemClick: (CategoryItem) -> Unit
) {
    Column(
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        items.chunked(2).forEach { rowItems ->
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                rowItems.forEach { item ->
                    CategoryCard(
                        item = item,
                        modifier = Modifier.weight(1f),
                        onClick = { onItemClick(item) }
                    )
                }
                // 如果只有1个item，添加一个空的Spacer来保持布局
                if (rowItems.size == 1) {
                    Spacer(modifier = Modifier.weight(1f))
                }
            }
        }
    }
}

@Composable
private fun CategoryCard(
    item: CategoryItem,
    modifier: Modifier = Modifier,
    onClick: () -> Unit
) {
    val themeColors = item.colors()
    val isDark = MaterialTheme.colorScheme.background.luminance < 0.5f // Use standard check

    // Premium Card Style (Matches Logic in LearningCalendarScreen)
    val containerColor = if (isDark) MaterialTheme.colorScheme.surfaceContainer else Color.White
    val borderColor = if (isDark) MaterialTheme.colorScheme.outlineVariant.copy(alpha = 0.05f) else MaterialTheme.colorScheme.outlineVariant.copy(alpha = 0.15f)
    val shadowElevation = if (isDark) 4.dp else 12.dp
    val shadowColor = if (isDark) Color.Black.copy(alpha = 0.5f) else Color.Black.copy(alpha = 0.06f)

    Surface(
        onClick = onClick,
        shape = RoundedCornerShape(26.dp),
        color = containerColor,
        border = BorderStroke(0.5.dp, borderColor),
        modifier = modifier
            .height(100.dp) // Adjusted height
            .shadow(
                elevation = shadowElevation,
                shape = RoundedCornerShape(26.dp),
                spotColor = shadowColor,
                ambientColor = shadowColor
            )
    ) {
        Row(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 20.dp, vertical = 16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Squircle Icon Container
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .clip(RoundedCornerShape(14.dp))
                    .background(themeColors.containerColor), // Pastel background
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = item.icon,
                    contentDescription = null,
                    modifier = Modifier.size(24.dp),
                    tint = themeColors.contentColor // Strong tint
                )
            }

            // Text
            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.Center
            ) {
                Text(
                    text = item.title,
                    style = MaterialTheme.typography.titleMedium.copy(
                        fontWeight = FontWeight.Bold,
                        fontSize = 17.sp
                    ),
                    color = MaterialTheme.colorScheme.onSurface, // Clean text color
                    maxLines = 1
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = item.subtitle,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.8f),
                    maxLines = 1
                )
            }
        }
    }
}

/**
 * 分类项数据类
 */
private data class CategoryItem(
    val id: String,
    val title: String,
    val subtitle: String,
    val icon: ImageVector,
    val colors: @Composable () -> CategoryThemeColor
)

// 扩展属性：获取Color的亮度
private val Color.luminance: Float
    get() = (0.299f * red + 0.587f * green + 0.114f * blue) // Keep for local usage

File: packages/features/feature_library/lib/src/presentation/category/category_classification_screen.dart
import 'package:core_designsystem/core_designsystem.dart';
import 'package:core_ui/core_ui.dart';
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import '../../routes/library_routes.dart';

/// 专项分类主题颜色
class _CategoryThemeColor {
  final Color containerColor;
  final Color contentColor;

  const _CategoryThemeColor({
    required this.containerColor,
    required this.contentColor,
  });
}

/// 专项训练/专项词汇 - 分类选择界面
/// 
/// 风格统一：
/// - Background: Solid (Theme background)
/// - Card: Premium Card (White/SurfaceContainer + Shadow + 26dp Rounded)
/// - Icon: Squircle (Rounded 14dp)
/// - Typography: ExtraBold Titles
class CategoryClassificationScreen extends StatelessWidget {
  const CategoryClassificationScreen({
    super.key,
    this.source = 'practice',
  });

  final String source;

  @override
  Widget build(BuildContext context) {

    final title = source == 'practice' ? '专项训练' : '专项词汇';

    return Scaffold(
      backgroundColor: NemoColors.bgBase,
      appBar: AppBar(
        title: Text(
          title,
          style: const TextStyle(fontWeight: FontWeight.w900),
        ),
        centerTitle: true,
        backgroundColor: Colors.transparent,
        elevation: 0,
        leading: IconButton(
          icon: const Icon(Icons.arrow_back_ios_new_rounded),
          onPressed: () => context.pop(),
        ),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 24),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            // 1. 基础词性类
            const _SectionHeader(title: '基础词性类'),
            const SizedBox(height: 16),
            _CategoryGrid(
              items: [
                _CategoryItem(
                  id: 'noun',
                  title: '名词类',
                  subtitle: '名词、代词等',
                  icon: Icons.menu_book_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardNounBgDark : NemoCategoryColors.cardNounBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardNounTextDark : NemoCategoryColors.cardNounTextLight,
                  ),
                ),
                _CategoryItem(
                  id: 'adj',
                  title: '形容词类',
                  subtitle: 'い形、な形形容词',
                  icon: Icons.stars_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardAdjIBgDark : NemoCategoryColors.cardAdjIBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardAdjITextDark : NemoCategoryColors.cardAdjITextLight,
                  ),
                ),
                _CategoryItem(
                  id: 'verb',
                  title: '动词类',
                  subtitle: '自动/他动/自他動词',
                  icon: Icons.directions_run_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardVerbBgDark : NemoCategoryColors.cardVerbBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardVerbTextDark : NemoCategoryColors.cardVerbTextLight,
                  ),
                ),
                _CategoryItem(
                  id: 'adv',
                  title: '副词',
                  subtitle: '修饰用言词汇',
                  icon: Icons.sort_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardAdvBgDark : NemoCategoryColors.cardAdvBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardAdvTextDark : NemoCategoryColors.cardAdvTextLight,
                  ),
                ),
              ],
              onItemClick: (item) => _onCategorySelected(context, item),
            ),

            const SizedBox(height: 32),

            // 2. 构词·句法功能类
            const _SectionHeader(title: '构词·句法功能类'),
            const SizedBox(height: 16),
            _CategoryGrid(
              items: [
                _CategoryItem(
                  id: 'rentai',
                  title: '连体词',
                  subtitle: '直接修饰体言',
                  icon: Icons.link_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardRentaiBgDark : NemoCategoryColors.cardRentaiBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardRentaiTextDark : NemoCategoryColors.cardRentaiTextLight,
                  ),
                ),
                _CategoryItem(
                  id: 'conj',
                  title: '接続词',
                  subtitle: '连接句子成分',
                  icon: Icons.linear_scale_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardConjBgDark : NemoCategoryColors.cardConjBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardConjTextDark : NemoCategoryColors.cardConjTextLight,
                  ),
                ),
                _CategoryItem(
                  id: 'exclam',
                  title: '感叹词',
                  subtitle: '表达情感语气',
                  icon: Icons.campaign_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardIdiomBgDark : NemoCategoryColors.cardIdiomBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardIdiomTextDark : NemoCategoryColors.cardIdiomTextLight,
                  ),
                ),
                _CategoryItem(
                  id: 'particle',
                  title: '助词',
                  subtitle: '语法功能标记',
                  icon: Icons.attribution_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardFixBgDark : NemoCategoryColors.cardFixBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardFixTextDark : NemoCategoryColors.cardFixTextLight,
                  ),
                ),
              ],
              onItemClick: (item) => _onCategorySelected(context, item),
            ),

            const SizedBox(height: 32),

            // 3. 构词·表达用法类
            const _SectionHeader(title: '构词·表达用法类'),
            const SizedBox(height: 16),
            _CategoryGrid(
              items: [
                _CategoryItem(
                  id: 'prefix',
                  title: '接头词',
                  subtitle: '词语前置构成',
                  icon: Icons.arrow_forward_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardKataBgDark : NemoCategoryColors.cardKataBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardKataTextDark : NemoCategoryColors.cardKataTextLight,
                  ),
                ),
                _CategoryItem(
                  id: 'suffix',
                  title: '接尾词',
                  subtitle: '词语后置构成',
                  icon: Icons.arrow_back_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardSoundBgDark : NemoCategoryColors.cardSoundBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardSoundTextDark : NemoCategoryColors.cardSoundTextLight,
                  ),
                ),
                _CategoryItem(
                  id: 'expression',
                  title: '表达·固定句型',
                  subtitle: '习惯表达方式',
                  icon: Icons.format_quote_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardKeigoBgDark : NemoCategoryColors.cardKeigoBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardKeigoTextDark : NemoCategoryColors.cardKeigoTextLight,
                  ),
                ),
                _CategoryItem(
                  id: 'kata',
                  title: '外来语',
                  subtitle: '片假名借词体系',
                  icon: Icons.language_rounded,
                  getTheme: (isDark) => _CategoryThemeColor(
                    containerColor: isDark ? NemoCategoryColors.cardAdjNaBgDark : NemoCategoryColors.cardAdjNaBgLight,
                    contentColor: isDark ? NemoCategoryColors.cardAdjNaTextDark : NemoCategoryColors.cardAdjNaTextLight,
                  ),
                ),
              ],
              onItemClick: (item) => _onCategorySelected(context, item),
            ),

            const SizedBox(height: 32),
          ],
        ),
      ),
    );
  }

  void _onCategorySelected(BuildContext context, _CategoryItem item) {
    if (source == 'practice') {
      // Use the named route 'learning-category' defined in LearningRoutes
      context.pushNamed(
        'learning-category',
        pathParameters: {'categoryId': item.id},
        queryParameters: {'title': item.title},
      );
    } else {
      context.pushNamed(
        LibraryRouteNames.categoryWords,
        pathParameters: {'categoryId': item.id},
        queryParameters: {'title': item.title},
      );
    }
  }
}

class _SectionHeader extends StatelessWidget {
  const _SectionHeader({required this.title});
  final String title;

  @override
  Widget build(BuildContext context) {
    return Text(
      title,
      style: Theme.of(context).textTheme.titleMedium?.copyWith(
            fontWeight: FontWeight.w900,
            letterSpacing: 0.5,
            color: Theme.of(context).colorScheme.onSurfaceVariant,
          ),
    );
  }
}

class _CategoryGrid extends StatelessWidget {
  const _CategoryGrid({
    required this.items,
    required this.onItemClick,
  });

  final List<_CategoryItem> items;
  final Function(_CategoryItem) onItemClick;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        for (var i = 0; i < items.length; i += 2)
          Padding(
            padding: const EdgeInsets.only(bottom: 16),
            child: Row(
              children: [
                Expanded(
                  child: _CategoryCard(
                    item: items[i],
                    onClick: () => onItemClick(items[i]),
                  ),
                ),
                const SizedBox(width: 16),
                if (i + 1 < items.length)
                  Expanded(
                    child: _CategoryCard(
                      item: items[i + 1],
                      onClick: () => onItemClick(items[i + 1]),
                    ),
                  )
                else
                  const Expanded(child: SizedBox()),
              ],
            ),
          ),
      ],
    );
  }
}

class _CategoryCard extends StatelessWidget {
  const _CategoryCard({
    required this.item,
    required this.onClick,
  });

  final _CategoryItem item;
  final VoidCallback onClick;

  @override
  Widget build(BuildContext context) {
    final theme = item.getTheme(Theme.of(context).brightness == Brightness.dark);

    return PremiumCard(
      padding: EdgeInsets.zero,
      borderRadius: BorderRadius.circular(26),
      child: Material(
        color: Colors.transparent,
        borderRadius: BorderRadius.circular(26),
        child: InkWell(
          borderRadius: BorderRadius.circular(26),
          onTap: onClick,
          child: Container(
            height: 100,
            padding: const EdgeInsets.symmetric(horizontal: 20),
            child: Row(
              children: [
                // Squircle Icon
                Container(
                  width: 48,
                  height: 48,
                  decoration: BoxDecoration(
                    color: theme.containerColor,
                    borderRadius: BorderRadius.circular(14),
                  ),
                  child: Icon(
                    item.icon,
                    color: theme.contentColor,
                    size: 24,
                  ),
                ),
                const SizedBox(width: 16),
                Expanded(
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        item.title,
                        style: const TextStyle(
                          fontWeight: FontWeight.w900,
                          fontSize: 17,
                        ),
                        maxLines: 1,
                        overflow: TextOverflow.ellipsis,
                      ),
                      const SizedBox(height: 4),
                      Text(
                        item.subtitle,
                        style: TextStyle(
                          fontSize: 12,
                          color: Theme.of(context).colorScheme.onSurfaceVariant.withValues(alpha: 0.8),
                          fontWeight: FontWeight.w600,
                        ),
                        maxLines: 1,
                        overflow: TextOverflow.ellipsis,
                      ),
                    ],
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

class _CategoryItem {
  final String id;
  final String title;
  final String subtitle;
  final IconData icon;
  final _CategoryThemeColor Function(bool isDark) getTheme;

  const _CategoryItem({
    required this.id,
    required this.title,
    required this.subtitle,
    required this.icon,
    required this.getTheme,
  });
}
